plugins/digitalocean/resoto_plugin_digitalocean/resources.py

```diff
@@ -61,14 +64,55 @@ def delete(self, graph: Graph) -> bool:
         raise NotImplementedError
 
     def update_tag(self, key, value) -> bool:
-        """Update a resource tag in the cloud"""
-        log.debug(f"Updating or setting tag {key}: {value} on resource {self.id}")
-        return True
+
+        if self.tag_resource_name():
+            log.debug(f"Updating tag {key} on resource {self.id}")
+            if value:
+                log.warning(
+                    "DigitalOcean does not support tag values. It will be ignored."
+                )
+            team = self._account
+            credentials = get_team_credentials(team.id)
+            client = StreamingWrapper(
+                credentials.api_token,
+                credentials.spaces_access_key,
+                credentials.spaces_secret_key,
+            )
+
+            tag_ready: bool = True
+
+            tag_count = client.get_tag_count(key)
+            # tag count call failed unrecoverably, we can't continue
+            if tag_count is None:
+                return False
+            # tag does not exist, create it
+            if tag_count == -1:
+                tag_ready = client.create_tag(key)
```

Comment:
mmh: I think raising an exception in case the `get_tag_count` fails unrecoverable would  be better? 
And None could indicate, that the tag does not exist?